lightning/src/ln/channelmanager.rs

```diff
@@ -1451,20 +1451,30 @@ macro_rules! remove_channel {
 
 macro_rules! send_channel_ready {
 	($self: ident, $pending_msg_events: expr, $channel: expr, $channel_ready_msg: expr) => {{
-		$pending_msg_events.push(events::MessageSendEvent::SendChannelReady {
-			node_id: $channel.get_counterparty_node_id(),
-			msg: $channel_ready_msg,
-		});
-		// Note that we may send a `channel_ready` multiple times for a channel if we reconnect, so
-		// we allow collisions, but we shouldn't ever be updating the channel ID pointed to.
-		let mut short_to_chan_info = $self.short_to_chan_info.write().unwrap();
-		let outbound_alias_insert = short_to_chan_info.insert($channel.outbound_scid_alias(), ($channel.get_counterparty_node_id(), $channel.channel_id()));
-		assert!(outbound_alias_insert.is_none() || outbound_alias_insert.unwrap() == ($channel.get_counterparty_node_id(), $channel.channel_id()),
-			"SCIDs should never collide - ensure you weren't behind the chain tip by a full month when creating channels");
-		if let Some(real_scid) = $channel.get_short_channel_id() {
-			let scid_insert = short_to_chan_info.insert(real_scid, ($channel.get_counterparty_node_id(), $channel.channel_id()));
-			assert!(scid_insert.is_none() || scid_insert.unwrap() == ($channel.get_counterparty_node_id(), $channel.channel_id()),
+		if $self.default_configuration.manually_signal_channel_ready {
+			let mut pending_events = $self.pending_events.lock().unwrap();
+			pending_events.push(events::Event::PendingChannelReady {
```

Comment:
ah, I didn't look carefully enough at `check_get_channel_ready` and didn't realize it can update channel state.

So the options are to update `check_get_channel_ready` so that like you said, it only returns the msg if manual is enabled and user has already signaled.  This implies adding state to a channel to track whether or not user has signaled already.

Is an alternative to update `check_get_channel_ready` so that it _never_ returns Some(msg) if manual signaling is enabled and then write new logic in the `signal_channel_readiness` to make sure it's safe to send and then builds and sends the message?

The latter approach would avoid adding additional state and is the way I'd lean going if you think it's okay?

Reply:
Hm true, I _was_ thinking it would just continue to fire Pending as needed, requiring the user to continually perform whatever checks they need before sending.  

I guess if we wanted to support only one approval from user then there's no way around adding state to the `Channel` so it knows whether or not it's received approval already

Reply:
I guess we can solve both problems at once and just add a `requires_manual_readiness_signal` to `Channel` that gets set to the new config at creation and then when they manually signal we can flip it to false.

Reply:
Just realized we also potentially send ChannelReady as part of channel reestablishment and monitor update workflows.  Seems like everywhere we might return Some(msg::ChannelReady) we would need to make sure it's been signaled already.

The issue then becomes where do we actually queue the PendingChannelReady event.  I guess the normal path is part of `do_chain_event` workflow?  Though maybe the first time can also be triggered after funding_created and a node restart via monitor update and 0-conf channel?

I think I need some help figuring out best place to queue it.